src/policy_artifacthub.rs

```diff
@@ -0,0 +1,923 @@
+                    KUBEWARDEN_ANNOTATION_POLICY_SOURCE,
+                    e
+                )
+            })?;
+            match policy_source.host_str() == Some("github.com") {
+                true => {
+                    let url = Url::parse(
+                        format!(
+                            "{}/releases/download/v{}/policy.wasm",
+                            policy_source.as_str(),
+                            version.to_string().as_str(),
+                        )
+                        .as_str(),
+                    )
+                    .map_err(|e| {
+                        anyhow!(
+                            "annotation \"{}\" in policy metadata must be a well formed url: {}",
+                            KUBEWARDEN_ANNOTATION_POLICY_SOURCE,
+                            e
+                        )
+                    })?;
+                    Ok(Some(vec![
+                        Link {
+                            name: ConstLinkName::Policy,
+                            url,
+                        },
+                        Link {
+                            name: ConstLinkName::Source,
+                            url: policy_source,
+                        },
+                    ]))
+                }
+                false => Ok(Some(vec![Link {
+                    name: ConstLinkName::Source,
+                    url: policy_source,
+                }])),
+            }
+        }
+        None => Ok(None),
+    }
+}
+
+// parses the value of annotation KUBEWARDEN_ANNOTATION_POLICY_AUTHOR into a
+// vector of maintainers, making sure the email is well formed, and removes '<'
+// and '>' from the mail address
+fn parse_maintainers(metadata_annots: &HashMap<String, String>) -> Result<Option<Vec<Maintainer>>> {
+    match metadata_annots.get(KUBEWARDEN_ANNOTATION_POLICY_AUTHOR) {
+        Some(s) => {
+            let mut maintainers: Vec<Maintainer> = vec![];
+            let csv = s
+                .split(',')
+                .map(|s| s.trim_start_matches(' '))
+                .map(|s| s.trim_end_matches(' '))
+                .map(str::to_string)
+                .collect::<Vec<String>>();
+            for word in csv.iter() {
+                if word.is_empty() {
+                    return Err(anyhow!(
+                        "annotation \"{}\" in policy metadata is malformed, must be csv values",
+                        KUBEWARDEN_ANNOTATION_POLICY_AUTHOR
+                    ));
+                }
+            }
+            for person in csv.iter() {
+                let vec_person = person.split('<').collect::<Vec<&str>>();
+                if vec_person.len() != 2 {
+                    return Err(anyhow!(
+                        "annotation \"{}\" in policy metadata is malformed, elements should be in \"name <email address>\" format",
+                        KUBEWARDEN_ANNOTATION_POLICY_AUTHOR
+                    ));
+                };
```

Comment:
Sadly, no.
I've tried to parse the `name-addr`, so, `Tux <tux@example.com>` according to https://www.rfc-editor.org/rfc/rfc5322#section-3.4) with:
   https://docs.rs/mail-parser/0.8.2/mail_parser/struct.Addr.html
and
  https://github.com/stalwartlabs/smtp-proto/blob/main/src/request/parser.rs#L401
  
 But without luck to be honest.
 I can keep looking a bit more, but I'm happy with the current approach: naively separate `display-name` and `angle-addr`, check the mail address inside `addr-spec`.

Reply:
I found a way to use `mail_parser`:

```rust
fn main() {
    let maintainers = "Homer Simpson <homer@simpson.com>, Marge Simpson <marge@simpson>";
    let to = format!("To: {maintainers}");

    let msg = mail_parser::Message::parse(to.as_bytes()).unwrap();

    let addr = msg.to();

    println!("Hello: {addr:?}");
}
```

The program produces this output:

```console
Hello: AddressList([Addr { name: Some("Homer Simpson"), address: Some("homer@simpson.com") }, Addr { name: Some("Marge Simpson"), address: Some("marge@simpson") }])
```

Reply:
still, that parser returns an Option, not a Result. It can contain a full mail message if the string is crafted on purpose.
Note that we still need the second lib, `email_address`, to parse the mail address itself, as `mail_parser` doesn't do it..
I keep looking.

Reply:
Done in https://github.com/kubewarden/policy-evaluator/pull/246/commits/9338445f091c3949a037e262fe5d2ed070ca2254.